scipy/stats/tests/test_multivariate.py

```diff
@@ -817,19 +826,54 @@ def test_2D_dirichlet_is_beta(self):
         np.random.seed(2846)
 
         alpha = np.random.uniform(10e-10, 100, 2)
-        d = dirichlet(alpha)
+        d = self.dist(alpha)
         b = beta(alpha[0], alpha[1])
 
         num_tests = 10
         for i in range(num_tests):
             x = np.random.uniform(10e-10, 100, 2)
             x /= np.sum(x)
-            assert_almost_equal(b.pdf(x), d.pdf([x]))
+            if self.dist == multivariate_beta:
+                assert_almost_equal(b.pdf(x), d.pdf(x[:, np.newaxis]))
+            else:
+                assert_almost_equal(b.pdf(x), d.pdf([x]))
 
         assert_almost_equal(b.mean(), d.mean()[0])
         assert_almost_equal(b.var(), d.var()[0])
 
 
+@pytest.mark.filterwarnings('ignore::DeprecationWarning')
+class TestDirichlet(DirichletTest):
+    dist = dirichlet
+
+
+def test_dirichlet_deprecation():
+    with pytest.deprecated_call():
+        dirichlet.rvs([1, 2, 3])
+
+    with pytest.deprecated_call():
+        dirichlet([1, 2, 3])
+
+
+class TestMultivariateBeta(DirichletTest):
+    dist = multivariate_beta
+
+    def test_accepts_random_variates(self):
+        # dirichlet.pdf does not accept output of dirichlet.rvs
+        # multivariate_beta does; see gh-6006
+        alpha = [1, 3, 4]
+        d = self.dist(alpha)
+
+        rvs = d.rvs()
+        pdf = d.pdf(rvs)  # no error
+        assert pdf == self.dist.pdf(rvs, alpha)
+
+        rvs2 = d.rvs(size=10)
+        pdf2 = d.pdf(rvs2)
+        for rvs_i, pdf_i in zip(rvs2, pdf2):
+            assert pdf_i == self.dist.pdf(rvs_i, alpha)
```

Comment:
are these `assert`s really necessary? We can just test if the call to pdf succeeds.

Reply:
I sort-of remember now - I wanted to test both the frozen and non-frozen versions. Maybe I figured that since I was calling both, I might as well check that they're identical, too?

Reply:
Oh, ok. It does cover the frozen distribution code. It's OK to keep them then.